src/components/Event/EventContainer.tsx

```diff
@@ -0,0 +1,99 @@
+import { useState } from "react";
+import EventBox, { type EventBoxProps } from "./EventBox";
+
+const Faculties = {
+  1: "Engineer",
+  2: "Science",
+  3: "Arts",
+};
+
+const EventContainer = ({ Events }: { Events: EventBoxProps[] }) => {
+  const [selectedFaculty, setSelectedFaculty] = useState<string | null>(null);
+  const [currEvents, setCurrEvents] = useState<EventBoxProps[]>(Events);
+  const [searchQuery, setSearchQuery] = useState<string>("");
+
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newSearchQuery = e.target.value;
+    setSearchQuery(newSearchQuery);
+
+    const filteredEvents = Events.filter((event) =>
+      event.EventName.toLowerCase().includes(newSearchQuery.toLowerCase())
+    );
+
+    const finalFilteredEvents =
+      selectedFaculty && selectedFaculty !== "คณะ / Faculty"
+        ? filteredEvents.filter(
+            (event) => event.EngFaculty === Faculties[selectedFaculty]
+          )
+        : filteredEvents;
+
+    setCurrEvents(finalFilteredEvents);
+  };
+
+  const handleFacultyChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedValue = e.target.value;
+    setSelectedFaculty(selectedValue);
+
+    // Filter events based on selected faculty
+    const filteredEvents =
+      selectedValue === "คณะ / Faculty"
+        ? Events
+        : Events.filter(
+            (event) => event.EngFaculty === Faculties[selectedValue]
+          );
+
+    const finalFilteredEvents =
+      searchQuery !== ""
+        ? filteredEvents.filter((event) =>
+            event.EventName.toLowerCase().includes(searchQuery.toLowerCase())
+          )
+        : filteredEvents;
```

Comment:
same to [readability and maintainability issue](https://github.com/isd-sgcu/oph66-frontend/pull/23/files#diff-d6a15aba8c64466da98664206bd709eee4ac1da6c8fd6b940d60d985f4e55aa1R23)